src/stream/double_ended/mod.rs

```diff
@@ -0,0 +1,201 @@
+mod next_back;
+mod nth_back;
+mod rfind;
+mod rfold;
+mod try_rfold;
+
+use next_back::NextBackFuture;
+use nth_back::NthBackFuture;
+use rfind::RFindFuture;
+use rfold::RFoldFuture;
+use try_rfold::TryRFoldFuture;
+
+extension_trait! {
+    use crate::stream::Stream;
+
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+
+    #[doc = r#"
+    Something fancy
+    "#]
+    pub trait DoubleEndedStream {
+        type Item;
+
+        fn poll_next_back(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>>;
+    }
+
+    #[doc = r#"
+        Something else
+    "#]
+    pub trait DoubleEndedStreamExt: crate::stream::DoubleEndedStream {
+        #[doc = r#"
+            Advances the stream and returns the next value.
+
+            Returns [`None`] when iteration is finished. Individual stream implementations may
+            choose to resume iteration, and so calling `next()` again may or may not eventually
+            start returning more values.
+
+            [`None`]: https://doc.rust-lang.org/std/option/enum.Option.html#variant.None
+
+            # Examples
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::stream::Sample;
+            use async_std::stream::double_ended::DoubleEndedStreamExt;
+
+            let mut s = Sample::from(vec![7u8]);
```

Comment:
Isn't it better to implement `DoubleEndedStream` in` FromIter`?
Was there any problem?